core/rawdb/accessors_state_sync.go

```diff
@@ -176,3 +176,19 @@ func NewSyncPerformedIterator(db ethdb.Iteratee) ethdb.Iterator {
 func UnpackSyncPerformedKey(key []byte) uint64 {
 	return binary.BigEndian.Uint64(key[len(syncPerformedPrefix):])
 }
+
+// GetLatestSyncPerformed returns the latest block number state synced performed to.
+func GetLatestSyncPerformed(db ethdb.Iteratee) uint64 {
+	// XXX: assuming this is not sorted
```

Comment:
hmm if they're sorted we probably won't even need to unpack all keys, but only the last key.

Since this is checked in startup only we can leave this as it is if we even have a slightest doubt about their ordering.

Reply:
The order is maintained by the db, let's just remove the `XXX` comment then?